website/content/docs/concepts/cpu.mdx

```diff
@@ -0,0 +1,317 @@
+client {
+  reserved {
+    cpu = 3000 # mhz
+  }
+}
+```
+
+Or the amount of reserved CPU can be specified by specifing a specific set of
+`cores` on which to disallow the scheduling of Nomad tasks. This capability is
+available on Linux systems only.
+
+```hcl
+client {
+  reserved {
+    cores = "0-3"
+  }
+}
+```
+
+Once the CPU is constrained by one of the above configurations, the node 
+attribute `cpu.usablecompute` will indicate the total amount of CPU bandwdith
+available for scheduling of Nomad tasks.
+
+## Allocating CPU Resources
+
+When scheduling a job a Task must specify how much CPU resource should be
+allocated on behalf of the task. This can be done in terms of bandwidth in MHz
+with the `cpu` attribute.
+
+```hcl
+task {
+  resources {
+    cpu = 2000 # mhz
+  }
+}
+```
+
+On Linux systems, Nomad supports reserving whole CPU cores specifically for a
+task. No task will be allowed to run on a CPU core reserved for another task.
+
+```hcl
+task {
+  resources {
+    cores = 4
+  }
+}
+```
+
+Nomad Enterprise supports NUMA aware scheduling, which enables operators to
+more finely control which CPU cores may be reserved for tasks.
+
+### CPU Hard Limits
+
+Some task drivers support the configuration option `cpu_hard_limit`. If enabled
+this option will restrict tasks from bursting above their CPU limit even
+when there is idle capacity on the node. The tradeoff is consistency versus
+utilization. A task with too few CPU resources may operate fine until another
+task is placed on the node causing a reduction in available CPU bandwidth, which
+could cause distruption for the underprovisioned task.
+
+### CPU Environment Variables
+
+To help tasks understand the resources available to them, Nomad sets environment
+variables in the runtime environment of each task.
+
+
+- `NOMAD_CPU_LIMIT` - The amount of CPU bandwidth allocated on behalf of the
+task.
+
+- `NOMAD_CPU_CORES` - The set of cores in [cpuset][] notation reserved for the task.
+This value is only set if `resources.cores` is configured.
+
+```env
+NOMAD_CPU_CORES=3-5
+NOMAD_CPU_LIMIT=9000
+```
+
+# NUMA
+
+Nomad clients are commonly provisioned on real hardware in an on-premise
+environment or in the cloud on large `.metal` instance types. In either case it
+is likely the underlying server is designed around a [NUMA topology][numa_wiki].
+Any server that contains multiple CPU sockets or multiple RAM banks per CPU
+socket will be characterized by the non-uniform access times involved in
+accessing system memory.
+
+
+[![NUMA](/img/nomad-numa.png)](/img/nomad-numa.png)
+
+The simplified example machine above has the following topology
+- 2 physical CPU sockets
+- 4 system memory banks, 2 per socket
+- 8 physical cpu cores (4 per socket)
+- 2 logical cpu cores per physical core
+- 4 PCI devices, 1 per memory bank
+
+### optimizing performance
+
+Any OS process scheduled on core X where X belongs to  memory node M will take
+longer to access any memory node that is not M.
```

Comment:
```suggestion
OS processes take longer to access memory nodes that don't belong to the core in which they're running.
```

Just to avoid some X and M variables .

Reply:
Ended up with

> Operating system processes take longer to access memory across a NUMA boundary.